### PR TITLE
Add spec URLs for HTML APIs

### PR DESCRIPTION
**Pull request by queengooborg**
This PR is a subset of #16051 that adds spec URLs for the HTML APIs.

### CODE REVIEW

File: api/HTMLElement.json [hunk 1]
```diff
@@ -420,6 +420,7 @@
       },
       "autocapitalize": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/interaction.html#dom-autocapitalize",
```
Comment (foolip) [inline api/HTMLElement.json:423]: There's a "For web developers" version of this one:

```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/interaction.html#dom-autocapitalize-dev",
```

I haven't checked all of them, but I suspect there are many more cases like this. I would suggest trying to append "-dev" to all URLs and checking if that URL is valid, and if so use that instead.

cc @sideshowbarker for other tips.

File: api/HTMLButtonElement.json [hunk 2]
```diff
@@ -147,6 +148,7 @@
       "form": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLButtonElement/form",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fae-form",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:151]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fae-form-dev",
```

File: api/HTMLButtonElement.json [hunk 3]
```diff
@@ -534,6 +542,7 @@
       "reportValidity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLButtonElement/reportValidity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-reportvalidity",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:545]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-reportvalidity-dev",
```

File: api/HTMLButtonElement.json [hunk 4]
```diff
@@ -581,6 +590,7 @@
       },
       "setCustomValidity": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-setcustomvalidity",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:593]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-setcustomvalidity-dev",
```

File: api/HTMLButtonElement.json [hunk 5]
```diff
@@ -677,6 +688,7 @@
       "validationMessage": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLButtonElement/validationMessage",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:691]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage-dev",
```

File: api/HTMLButtonElement.json [hunk 6]
```diff
@@ -725,6 +737,7 @@
       "validity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLButtonElement/validity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validity",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:740]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#the-constraint-validation-api:dom-cva-validity",
```

File: api/HTMLButtonElement.json [hunk 7]
```diff
@@ -821,6 +835,7 @@
       "willValidate": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLButtonElement/willValidate",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate",
```
Comment (sideshowbarker) [inline api/HTMLButtonElement.json:838]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate-dev",
```

File: api/HTMLDataListElement.json [hunk 8]
```diff
@@ -51,6 +51,7 @@
       "options": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLDataListElement/options",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-datalist-options",
```
Comment (sideshowbarker) [inline api/HTMLDataListElement.json:54]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-datalist-options-dev",
```

File: api/HTMLElement.json [hunk 9]
```diff
@@ -2836,6 +2838,7 @@
       "spellcheck": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLElement/spellcheck",
+          "spec_url": "https://html.spec.whatwg.org/multipage/interaction.html#dom-spellcheck",
```
Comment (sideshowbarker) [inline api/HTMLElement.json:2841]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/interaction.html#dom-spellcheck-dev",
```

File: api/HTMLEmbedElement.json [hunk 10]
```diff
@@ -103,6 +104,7 @@
       },
       "getSVGDocument": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/embedded-content-other.html#dom-media-getsvgdocument",
```
Comment (sideshowbarker) [inline api/HTMLEmbedElement.json:107]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/embedded-content-other.html#dom-media-getsvgdocument-dev",
```

File: api/HTMLFieldSetElement.json [hunk 11]
```diff
@@ -50,6 +50,7 @@
       },
       "checkValidity": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-checkvalidity",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:53]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-checkvalidity-dev",
```

File: api/HTMLFieldSetElement.json [hunk 12]
```diff
@@ -146,6 +148,7 @@
       "elements": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFieldSetElement/elements",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-fieldset-elements",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:151]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-fieldset-elements-dev",
```

File: api/HTMLFieldSetElement.json [hunk 13]
```diff
@@ -338,6 +343,7 @@
       "reportValidity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFieldSetElement/reportValidity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-reportvalidity",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:346]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-reportvalidity-dev",
```

File: api/HTMLFieldSetElement.json [hunk 14]
```diff
@@ -385,6 +391,7 @@
       },
       "setCustomValidity": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-setcustomvalidity",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:394]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-setcustomvalidity-dev",
```

File: api/HTMLFieldSetElement.json [hunk 15]
```diff
@@ -433,6 +440,7 @@
       "type": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFieldSetElement/type",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-fieldset-type",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:443]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-fieldset-type-dev",
```

File: api/HTMLFieldSetElement.json [hunk 16]
```diff
@@ -481,6 +489,7 @@
       "validationMessage": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFieldSetElement/validationMessage",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:492]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage-dev",
```

File: api/HTMLFieldSetElement.json [hunk 17]
```diff
@@ -577,6 +587,7 @@
       "willValidate": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFieldSetElement/willValidate",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate",
```
Comment (sideshowbarker) [inline api/HTMLFieldSetElement.json:590]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate-dev",
```

File: api/HTMLFormElement.json [hunk 18]
```diff
@@ -197,6 +198,7 @@
       "checkValidity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLFormElement/checkValidity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/forms.html#dom-form-checkvalidity",
```
Comment (sideshowbarker) [inline api/HTMLFormElement.json:201]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/forms.html#dom-form-checkvalidity-dev",
```

File: api/HTMLIFrameElement.json [hunk 19]
```diff
@@ -589,6 +593,7 @@
       },
       "getSVGDocument": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/embedded-content-other.html#dom-media-getsvgdocument",
```
Comment (sideshowbarker) [inline api/HTMLIFrameElement.json:596]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/embedded-content-other.html#dom-media-getsvgdocument-dev",
```

File: api/HTMLInputElement.json [hunk 20]
```diff
@@ -2992,6 +3037,8 @@
       },
       "validationMessage": {
         "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLInputElement/validationMessage",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage",
```
Comment (sideshowbarker) [inline api/HTMLInputElement.json:3041]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validationmessage-dev",
```

File: api/HTMLInputElement.json [hunk 21]
```diff
@@ -3133,6 +3183,7 @@
       },
       "valueAsDate": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasdate",
```
Comment (sideshowbarker) [inline api/HTMLInputElement.json:3186]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasdate-dev",
```

File: api/HTMLInputElement.json [hunk 22]
```diff
@@ -3192,6 +3243,7 @@
       },
       "valueAsNumber": {
         "__compat": {
+          "spec_url": "https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasnumber",
```
Comment (sideshowbarker) [inline api/HTMLInputElement.json:3246]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/input.html#dom-input-valueasnumber-dev",
```

File: api/HTMLInputElement.json [hunk 23]
```diff
@@ -3384,6 +3438,8 @@
       },
       "willValidate": {
         "__compat": {
+          "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLInputElement/willValidate",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate",
```
Comment (sideshowbarker) [inline api/HTMLInputElement.json:3442]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate-dev",
```

File: api/HTMLMapElement.json [hunk 24]
```diff
@@ -51,6 +51,7 @@
       "areas": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLMapElement/areas",
+          "spec_url": "https://html.spec.whatwg.org/multipage/image-maps.html#dom-map-areas",
```
Comment (sideshowbarker) [inline api/HTMLMapElement.json:54]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/image-maps.html#dom-map-areas-dev",
```

File: api/HTMLMediaElement.json [hunk 25]
```diff
@@ -101,6 +101,7 @@
       "addTextTrack": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLMediaElement/addTextTrack",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-media-addtexttrack",
```
Comment (sideshowbarker) [inline api/HTMLMediaElement.json:104]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-media-addtexttrack-dev",
```

File: api/HTMLMediaElement.json [hunk 26]
```diff
@@ -2511,6 +2514,7 @@
       "played": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLMediaElement/played",
+          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-media-played",
```
Comment (sideshowbarker) [inline api/HTMLMediaElement.json:2517]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/media.html#dom-media-played-dev",
```

File: api/HTMLOptionElement.json [hunk 27]
```diff
@@ -198,6 +200,7 @@
       "form": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionElement/form",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-form",
```
Comment (sideshowbarker) [inline api/HTMLOptionElement.json:203]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-form-dev",
```

File: api/HTMLOptionElement.json [hunk 28]
```diff
@@ -246,6 +249,7 @@
       "index": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionElement/index",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-index",
```
Comment (sideshowbarker) [inline api/HTMLOptionElement.json:252]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-index-dev",
```

File: api/HTMLOptionElement.json [hunk 29]
```diff
@@ -358,6 +363,7 @@
       "selected": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionElement/selected",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-selected",
```
Comment (sideshowbarker) [inline api/HTMLOptionElement.json:366]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-selected-dev",
```

File: api/HTMLOptionElement.json [hunk 30]
```diff
@@ -406,6 +412,7 @@
       "text": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionElement/text",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-text",
```
Comment (sideshowbarker) [inline api/HTMLOptionElement.json:415]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-option-text-dev",
```

File: api/HTMLOptionsCollection.json [hunk 31]
```diff
@@ -51,6 +51,7 @@
       "add": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionsCollection/add",
+          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-add",
```
Comment (sideshowbarker) [inline api/HTMLOptionsCollection.json:54]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-add-dev",
```

File: api/HTMLOptionsCollection.json [hunk 32]
```diff
@@ -147,6 +149,7 @@
       "remove": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionsCollection/remove",
+          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-remove",
```
Comment (sideshowbarker) [inline api/HTMLOptionsCollection.json:152]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-remove-dev",
```

File: api/HTMLOptionsCollection.json [hunk 33]
```diff
@@ -195,6 +198,7 @@
       "selectedIndex": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOptionsCollection/selectedIndex",
+          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#dom-htmloptionscollection-selectedindex",
```
Comment (sideshowbarker) [inline api/HTMLOptionsCollection.json:201]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/common-dom-interfaces.html#the-htmloptionscollection-interface:dom-htmloptionscollection-selectedindex-2",
```

File: api/HTMLOutputElement.json [hunk 34]
```diff
@@ -100,6 +100,7 @@
       "checkValidity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/checkValidity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-checkvalidity",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:103]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-checkvalidity-dev",
```

File: api/HTMLOutputElement.json [hunk 35]
```diff
@@ -148,6 +149,7 @@
       "defaultValue": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/defaultValue",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-defaultvalue",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:152]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-defaultvalue-dev",
```

File: api/HTMLOutputElement.json [hunk 36]
```diff
@@ -196,6 +198,7 @@
       "form": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/form",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fae-form",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:201]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-fae-form-dev",
```

File: api/HTMLOutputElement.json [hunk 37]
```diff
@@ -491,6 +498,7 @@
       "type": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/type",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-type",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:501]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-type-dev",
```

File: api/HTMLOutputElement.json [hunk 38]
```diff
@@ -587,6 +596,7 @@
       "validity": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/validity",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-validity",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:599]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#the-constraint-validation-api:dom-cva-validity",
```

File: api/HTMLOutputElement.json [hunk 39]
```diff
@@ -635,6 +645,7 @@
       "value": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/value",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-value",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:648]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-output-value-dev",
```

File: api/HTMLOutputElement.json [hunk 40]
```diff
@@ -683,6 +694,7 @@
       "willValidate": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLOutputElement/willValidate",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate",
```
Comment (sideshowbarker) [inline api/HTMLOutputElement.json:697]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#dom-cva-willvalidate-dev",
```

File: api/HTMLProgressElement.json [hunk 41]
```diff
@@ -148,6 +149,7 @@
       "position": {
         "__compat": {
           "mdn_url": "https://developer.mozilla.org/docs/Web/API/HTMLProgressElement/position",
+          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-progress-position",
```
Comment (sideshowbarker) [inline api/HTMLProgressElement.json:152]:
```suggestion
          "spec_url": "https://html.spec.whatwg.org/multipage/form-elements.html#dom-progress-position-dev",
```

**Pull request merged**